core/dataset_operations_common.h

```diff
@@ -7,6 +7,20 @@
 
 namespace scipp::core {
 
+/// Return the dimension for given coord or labels.
+///
+/// For dimension-coords, this is the same as the key, for non-dimension-coords
+/// (labels) we adopt the convention that labels are "labelling" their inner
+/// dimension.
+template <class T, class Key>
+Dim dim_of_coord_or_labels(const T &var, const Key &key) {
+  if constexpr (std::is_same_v<Key, Dim>) {
```

Comment:
Just wondering how you can still have a key that is not a `Dim`, if they are all the same for dimension coords and non-dimension coords?

Reply:
This is because `labels` is not removed yet. The check if the key-type will be removed alongside the future removal of the `labels` properties.

Reply:
(the `else` will be removed)